db/index.ts

```diff
@@ -28,23 +29,187 @@ CREATE TABLE IF NOT EXISTS settings (
 );
 `;
 
-export const expo = SQLite.openDatabaseSync('period.db');
+let expo: SQLite.SQLiteDatabase | null = null;
+let db: ReturnType<typeof drizzle> | null = null;
+let initializationPromise: Promise<void> | null = null;
```

Comment:
I really don't see the point of this sort of promises I'm seeing in this PR. They make everything so complicated. Is it to offer some kind of mutex? If so, use a real mutex if you really need it.